mongodb/spring-cloud-starter-stream-source-mongodb/src/main/java/org/springframework/cloud/stream/app/mongodb/source/MongoDBSourceProperties.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2016 the original author or authors.
+ * Licensed under the Apache License, Version 2.0 (the "License") ;
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *          http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.app.mongodb.source;
+
+import org.hibernate.validator.constraints.NotBlank;
+import org.hibernate.validator.constraints.NotEmpty;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+/**
+ * @author Adam Zwickey
+ *
+ */
+@ConfigurationProperties
+public class MongoDBSourceProperties {
+
+    /**
+     * The MongoDB collection to query
+     */
+    private String collection;
+    /**
+     * The MongoDB query
+     */
+    private String query = "{ }";
```

Comment:
I wonder if we should consider some `updateQuery` option. Fully similar to what we have with JDBC Source and it is really logical.
Right, I agree that `MongoDbMessageSource` doesn't provide such an option, but from other side it exposes its `mongoTemplate` to the `TransactionSynchronizationManager` resource. So, we can do something like this:
````xml
<int-mongodb:inbound-channel-adapter channel="input"
								query="{'name' : 'Bob'}">
	<int:poller fixed-rate="200" max-messages-per-poll="1">
	    <int:transactional synchronization-factory="syncFactory"/>
	</int:poller>
</int-mongodb:inbound-channel-adapter>
	
<int:transaction-synchronization-factory id="syncFactory">
	<int:after-commit expression="@documentCleaner.remove(#mongoTemplate, payload, headers.mongo_collectionName)"/>
</int:transaction-synchronization-factory>
````
Pay attention to the `syncFactory` definition. That is how we can come up with the `update` option do not pick up the same documents on the next poll.

OTOH we can consider this improvement in the future. After all Spring XD lives without an update anyway.
WDYT @ericbottard ?